components/Homepage/RoundActionCard/RoundActionCard.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardBody,
+  Center,
+  HStack,
+  Heading,
+  Spinner,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+import { Phase } from "services/PhaseMgmtService";
+
+interface Props {
+  phase: Phase;
+  roundId: number;
+  isAuthed?: boolean;
+  hasSignedUp?: boolean;
+  hasVoted?: boolean;
+  hasSubmitted?: boolean;
+  onProfile: () => void;
+  onSignup: () => void;
+  onSignupAndJoinRound: () => void;
+  onJoinRound: () => void;
+  onVote: () => void;
+  onSubmit: () => void;
+  onRoundDetails: () => void;
+  loading?: boolean;
+}
+
+export const RoundActionCard = ({
+  phase,
+  roundId,
+  isAuthed,
+  hasSignedUp,
+  hasSubmitted,
+  hasVoted,
+  onProfile,
+  onSignup,
+  onSignupAndJoinRound,
+  onJoinRound,
+  onVote,
+  onSubmit,
+  onRoundDetails,
+  loading,
+}: Props) => {
+  const renderLabelContent = () => {
+    if (isAuthed) {
+      switch (phase) {
+        case "signups":
+          return <>Next round starts in X days</>;
+        case "celebration":
+          return <>Stay tuned for next round details!</>;
+        case "voting":
+          return <>Voting ends in X days</>;
+        case "covering":
+          return <>Round ends in X days</>;
+      }
+    } else {
+      if (phase === "signups") {
+        return <>Next round starts in X days</>;
+      } else {
+        return <>Notify me when next round starts</>;
+      }
+    }
+  };
+
+  const renderCTAContent = () => {
+    if (!isAuthed) {
```

Comment:
nit - From a readability standpoint, since these all render a button and optional text, can all the business logic render those 3 variables - button text, onClick function and the text, and then at the end, actually render the JSX? Instead of each logic block rendering JSX? This just seems unnecessarily verbose.